core/network/src/on_demand.rs

```diff
@@ -316,6 +324,45 @@ impl<B> OnDemandService<B> for OnDemand<B> where
 			data => Accept::Unexpected(data),
 		})
 	}
+
+	fn on_remote_body_response(&self, peer: PeerId, response: message::BlockResponse<B>) {
+		self.accept_response("body", peer, response.id, |request| match request.data {
+			RequestData::RemoteBody(request, sender) => {
+				let num_bodies = response.blocks.iter().count();
+
+				// Number of bodies are hardcoded to 1 for valid `RemoteBodyResponses`
+				if num_bodies != 1 {
+					return Accept::CheckFailed("RemoteBodyResponse: invalid number of blocks".into(),
+												RequestData::RemoteBody(request, sender))
+				}
+
+				let response = response
+					.blocks
+					.into_iter()
+					.take(1)
```

Comment:
This `.take(1)` does noting, if I understand correctly.

Reply:
yes, thanks